### PR TITLE
Avoid re-issuing redundant cross-references.

### LINKED ISSUE #8722: [impovement] don't reverence same issue multible times next to itself
just look at: https://gitea.com/gitea/tea/pulls/45
Gitea Version: 1.11.0+dev-56-g28f60bb5c

Bildschirmfoto zu 2019-10-28 22-33-12

### PR DESCRIPTION
**Pull request by guillep2k**
Fixes #8722 

This will avoid rewriting the reference if its the same as an existing one.

### CODE REVIEW

**Comment by 6543** [comment]
I'll test it this day ...

File: models/issue_xref.go [hunk 1]
```diff
@@ -53,13 +54,23 @@ func neuterCrossReferences(e Engine, issueID int64, commentID int64) error {
 	if commentID != 0 {
 		sess = sess.And("`ref_comment_id` = ?", commentID)
 	}
-	if err := sess.Find(&active); err != nil || len(active) == 0 {
+	err := sess.Find(&active)
```
Comment (lafriks) [inline models/issue_xref.go]:
```suggestion
```

File: models/issue_xref.go [hunk 2]
```diff
@@ -53,13 +54,23 @@ func neuterCrossReferences(e Engine, issueID int64, commentID int64) error {
...
+	return active, err
```
Comment (lafriks) [inline models/issue_xref.go]:
```suggestion
	return active, sess.Find(&active)
```

**Review by lafriks — CHANGES_REQUESTED** [review]

File: models/issue_xref.go (hunk 2 above)
Comment (guillep2k) [reply models/issue_xref.go]: @lafriks done

**Comment by guillep2k** [comment]
In theory this could be ported to 1.10.0.

**Comment by 6543** [comment]
this case does still produce redundancy:
```sh
create issue A
create issue B

from B reverence A
from B reverence A again
from B reverence A again
```
Result: same issue referenced in line ...

what i expect until there is a message in between no new reverence/update last rev

**Comment by guillep2k** [comment]
> 
> 
> this case does still produce redundancy:
> 
> ```shell
> create issue A
> create issue B
> 
> from B reverence A
> from B reverence A again
> from B reverence A again
> ```
> 
> Result: same issue referenced in line ...
> 
> what i expect until there is a message in between no new reverence/update last rev

I don't think I should remove that case. If the issue was referenced twice, IMHO there should be two lines with references. The md parsing code will eliminate references from quotes and code, so I think it's worth leaving them.

Even from the algorithm point of view, think of what would happen if the users decides to delete the first comment? Should I parse all comments and check again all possible references before I delete the corresponding one?

Besides, the reference is not the same: they each point to a different comment#snug

**Review by 6543 — APPROVED** [review]

**Comment by 6543** [comment]
@lafriks can you review?

**Review by lafriks — APPROVED** [review]

**Comment by 6543** [comment]
@guillep2k can you resolve conflicts thx

**Comment by guillep2k** [comment]
Conflicts solved

**Comment by lafriks** [comment]
Make LG-TM work :)

**Pull request merged**